backend/msgpack/server.go

```diff
@@ -182,12 +194,45 @@ func (s *server) write(
 		s.RUnlock()
 		return err
 	}
-	numShards := placement.NumShards()
-	shard := s.shardFn(id, numShards)
-	instances := placement.InstancesForShard(shard)
-	err = s.writerMgr.WriteTo(instances, shard, mu, pl)
+	var (
+		numShards = placement.NumShards()
+		shardID   = s.shardFn(id, numShards)
+		instances = placement.InstancesForShard(shardID)
```

Comment:
Yeah the placement is dynamically determined based on the current time and the cutover time of each placement in the list of staged placements. See https://github.com/m3db/m3cluster/blob/master/services/placement/staged_placement.go#L77.